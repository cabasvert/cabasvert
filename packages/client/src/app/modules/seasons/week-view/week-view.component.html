<ng-container *ngIf="(seasonWeek$ | async) as week">
  {{ week.distributionDate | date: 'shortDate' }}
  ({{ week.seasonWeek }})
</ng-container>
