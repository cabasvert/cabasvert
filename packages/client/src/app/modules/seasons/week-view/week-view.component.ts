/**
 * This file is part of CabasVert.
 *
 * Copyright 2017, 2018 Didier Villevalois
 *
 * CabasVert is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * CabasVert is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with CabasVert.  If not, see <http://www.gnu.org/licenses/>.
 */

import { Component, Input, OnInit } from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { SeasonWeek } from '../season.model';
import { SeasonService } from '../season.service';

@Component({
  selector: 'app-week-view',
  templateUrl: './week-view.component.html',
  styleUrls: ['./week-view.component.scss'],
})
export class WeekViewComponent implements OnInit {

  @Input('seasonId') seasonId: string;
  @Input('weekNumber') weekNumber: number;

  seasonWeek$: Observable<SeasonWeek>;

  constructor(private seasonService: SeasonService) {
  }

  ngOnInit() {
    this.seasonWeek$ = this.seasonService.seasonById$(this.seasonId).pipe(
      map(season => season.seasonWeekByNumber(this.weekNumber)),
    );
  }
}
