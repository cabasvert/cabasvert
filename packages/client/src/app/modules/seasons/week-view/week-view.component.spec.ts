/*
 * This file is part of CabasVert.
 *
 * Copyright 2017, 2018 Didier Villevalois
 *
 * CabasVert is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * CabasVert is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with CabasVert.  If not, see <http://www.gnu.org/licenses/>.
 */

import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { of } from 'rxjs'
import { SeasonService } from '../season.service'

import { WeekViewComponent } from './week-view.component'

describe('WeekViewComponent', () => {
  let seasonService

  beforeEach(async(() => {
    seasonService = {
      seasonById$: jest.fn(),
    }

    TestBed.configureTestingModule({
        declarations: [WeekViewComponent],
        providers: [
          { provide: SeasonService, useValue: seasonService },
        ],
      })
      .compileComponents()
  }))

  let component: WeekViewComponent
  let fixture: ComponentFixture<WeekViewComponent>

  beforeEach(() => {
    fixture = TestBed.createComponent(WeekViewComponent)
    component = fixture.componentInstance
  })

  it('should create', () => {
    expect(component).toBeTruthy()
  })

  it('should display the week', () => {
    let seasonId = 'season:2018S'
    let weekNumber = 7

    let season = {
      seasonWeekByNumber: jest.fn(),
    }
    seasonService.seasonById$.mockReturnValue(of(season))

    let week = { distributionDate: new Date(2018, 3, 1), seasonWeek: 7 }
    season.seasonWeekByNumber.mockReturnValue(week)

    component.seasonId = seasonId
    component.weekNumber = weekNumber

    component.ngOnChanges()
    fixture.detectChanges()

    expect(seasonService.seasonById$).toBeCalledWith(seasonId)
    expect(season.seasonWeekByNumber).toBeCalledWith(weekNumber)

    expect(season.seasonWeekByNumber).toHaveBeenCalled()

    let content = fixture.nativeElement.textContent
    expect(content).toContain('4/1/18 (7)')
  })
})
