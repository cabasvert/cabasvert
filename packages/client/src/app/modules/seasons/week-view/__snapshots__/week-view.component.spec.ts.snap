// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekViewComponent should display the week 1`] = `
<app-week-view
  seasonId={[Function String]}
  seasonService={[Function Object]}
  seasonWeek$={[Function Observable]}
  weekNumber={[Function Number]}
>
   4/1/18 (7)
  
</app-week-view>
`;
